main.js

```diff
@@ -1,96 +1,6 @@
-var onlyKey = new OnlyKey();
```

Comment:
Without instantiating this "lite" version of the OnlyKey class in main.js, users have to launch the app to set the time on their OnlyKey, which is required for properly syncing 2FA codes. This was a kludgey way to fire OKSETTIME in a Chrome background process.

Reply:
With the NW.js app we probably don't need to use the background page since this is a standalone app its not going to be started automatically by Chrome right? If not then OKSETTIME will just fire whenever it detects an OnlyKey connected. At some point we should try to make it so that the the NW.js app starts automatically with OS.

Reply:
Yeah, but we can't merge in this change to the master branch of our existing app. As I recall, we are looking for a way to essentially run the existing app within nw.js (or Electron, or any native OS app wrapper) "AS-IS," even pulling updates from the Chrome Web Store if possible, to reduce maintenance and confusion. This would break our existing app if merged in and published.

Reply:
True lets keep support for Chrome apps. Even after they are discontinued we want to keep support for this app as Chrome apps are not being discontinued for Chromebook. Let's keep the chrome app stuff but check to see if its a chrome app - this might work - https://stackoverflow.com/questions/7507277/detecting-if-code-is-being-run-as-a-chrome-extension

Reply:
OK, I'll leave this in.

Just to make sure I understand this... Is this running anytime Chrome is running? And does this mean TOTP only works when the user has installed the app and is running Chrome?

Also, the key would get two connections and two OKSETTIME messages when it is plugged in while the app is running. Is this correct and does the key handle this?

Reply:
@Sjlver The way TOTP (time based OTP) works is that the OTP is generated based on a shared secret and the current time. This way both client and server can verify they have the same OTP and it changes every 30 seconds. This requires the OnlyKey to know the time, a device without a battery cannot  keep time. So yes, if you try to use OnlyKey on a system that does not have the OnlyKey app installed you will see that the OTP prints "NOTSET" instead of the OTP. The app uses a background page which is basically automatically runs on system start through Chrome, this background page sends the time to OnlyKey whenever it detects an OnlyKey. Onlykey handles OKSETTIME messages by setting the time to the time of the first OKSETTIME message it receives, since its powered it can then keep good time after that. Additionally, OKSETTIME messages are used as a Status/Ping type of message for the app, so you will notice that the OnlyKey responds to OKSETTIME with Unlocked and firmware version # if it is unlocked, Initialized if the OnlyKey is locked, and Uninitialized if no PIN has ever been set on OnlyKey. This is how the app knows to present the user with either the initial setup screen or the slot configuration screen.

Reply:
Thanks for the explanation!